xamarin/Xamarin.Forms
# [ios] Shell inset,  general SafeArea fixes for shell and non shell embedded pages, entry scrolling fixes

### Description of Change ###

- Fix iOS so the content pads correctly for: pages with top tabs, all pages not just scroll views
- Fixed the keyboard focus auto scroll so that when you click off the entry it'll scroll back correctly
- fixed safe area so that if you turn it off content will set to 0,0 (this also applies to page with top tabs)
- fixed safe area padding to include top tabs size as well
- consolidated all the safe area setting logic to a single method so it's easy to follow the logic of why we set it when

### Issues Resolved ### 
- fixes #5239
- fixes #6424
- fixes #6499

### Platforms Affected ### 
- iOS

### Breaking ###

- prior to this PR if the user set the padding, set safearea to true, then set the safearea to false it would reset the padding to zero. After this PR it'll set the padding back to whatever the user initially set it to


### Testing Procedure ###
I added a ShellInsets test that has some automation but you can just click around on things to see if it all works to your liking as well

### PR Checklist ###
